packages/medusa-file-s3/src/services/s3.ts

```diff
@@ -63,65 +76,68 @@ class S3Service extends AbstractFileService implements IFileService {
       acl: undefined,
     }
   ) {
-    const client = this.getClient()
 
     const parsedFilename = parse(file.originalname)
 
     const fileKey = `${parsedFilename.name}-${Date.now()}${parsedFilename.ext}`
 
-    const params = {
+    const command = new PutObjectCommand({
       ACL: options.acl ?? (options.isProtected ? "private" : "public-read"),
       Bucket: this.bucket_,
       Body: fs.createReadStream(file.path),
       Key: fileKey,
       ContentType: file.mimetype,
-    }
-
-    const result = await client.upload(params).promise()
+      CacheControl: this.cacheControl_
+    })
 
-    return {
-      url: result.Location,
-      key: result.Key,
+    try {
+      await this.client_.send(command)
+      return {
+        url: `${this.s3Url_}/${fileKey}`,
+        key: fileKey,
+      }
+    } catch (e) {
+      this.logger_.error(e)
+      throw(e) // rejects the promise
     }
   }
 
   async delete(file: DeleteFileType): Promise<void> {
-    const client = this.getClient()
-
-    const params = {
+   const command = new DeleteObjectCommand({
       Bucket: this.bucket_,
-      Key: `${file}`,
-    }
-
-    return new Promise((resolve, reject) => {
-      client.deleteObject(params, (err, data) => {
-        if (err) {
-          reject(err)
-          return
-        }
-        resolve()
-      })
+      Key: `${file.file_key}`,
     })
+
+    try {
+      const response = await this.client_.send(command)
+      console.log(response)
+    } catch (e) {
+      this.logger_.error(e)
+    }
   }
 
   async getUploadStreamDescriptor(fileData: UploadStreamDescriptorType) {
-    const client = this.getClient()
     const pass = new stream.PassThrough()
 
     const isPrivate = fileData.isPrivate ?? true // default to private
 
     const fileKey = `${fileData.name}.${fileData.ext}`
-    const params: PutObjectRequest = {
+    const params: PutObjectCommandInput = {
       ACL: isPrivate ? "private" : "public-read",
       Bucket: this.bucket_,
       Body: pass,
       Key: fileKey,
       ContentType: fileData.contentType as string,
     }
 
+    const uploadJob = new Upload({
+      client: this.client_,
+      params
+    })
+
     return {
       writeStream: pass,
-      promise: client.upload(params).promise(),
+      promise: uploadJob.done(),
```

Comment:
question: Curious, why are we not following a similar command pattern as with delete and update?

Note, I haven't read through the v3 documentation, so apologies in advance for my ignorance :)

Reply:
The current version of the plugin was already set up to use a stream passthrough for this function but not in the default upload function.  My guess was that it was done that way because the import/export files might be very large.

I took the approach of using the @aws-sdk/lib-storage helper/companion package to handle the stream since it is designed to handle buffers, blobs, streams, or multipart uploads.  I don't think it's necessary in the regular upload function.

Here is more info: https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/modules/_aws_sdk_lib_storage.html